bfvmm/src/vmcs/test/test_vmcs_intel_x64.cpp

```diff
@@ -871,17 +1041,28 @@ vmcs_ut::test_vmcs_guest_tr_selector_index()
 
     vmcs::guest_tr_selector::index::set(0UL);
     this->expect_true(vmcs::guest_tr_selector::index::get() == 0UL);
+
+    vmcs::guest_tr_selector::index::set_if_exists(1UL);
+    this->expect_true(vmcs::guest_tr_selector::index::get_if_exists() == 1UL);
+
+    vmcs::guest_tr_selector::index::set_if_exists(0UL);
+    this->expect_true(vmcs::guest_tr_selector::index::get_if_exists() == 0UL);
 }
 
 void
 vmcs_ut::test_vmcs_guest_interrupt_status()
 {
-    vmcs::guest_interrupt_status::set(100UL);
-    g_msrs[msrs::ia32_vmx_procbased_ctls2::addr] = msrs::ia32_vmx_procbased_ctls2::virtual_interrupt_delivery::mask;
+    g_msrs[msrs::ia32_vmx_true_procbased_ctls::addr] = msrs::ia32_vmx_true_procbased_ctls::activate_secondary_controls::mask << 32;
```

Comment:
if exists relies on the secondary controls, we should be testing when they are disabled as well as enabled, and the same goes for the interrupt status. 

Please review other fields and make sure that these same tests are not missing